src/bucket/Bucket.cpp

```diff
@@ -855,7 +873,7 @@ Bucket::scanForEviction(AbstractLedgerTxn& ltx, EvictionIterator& iter,
         return true;
     }
 
-    auto& stream = getStream();
+    auto& stream = getEvictionStream();
     stream.seek(iter.bucketFileOffset);
```

Comment:
isn't this `seek` call redundant given that eviction stream should now guarantee position validity? (we should probably also assert that bucketFileOffset is consistent with stream pos)

Reply:
No, if a node is catching up or joining the network, the stream will need to be seeked to the correct position based on the config parameter. I'd rather error on the safer side and re-seek every time instead of special casing an init path.

Reply:
Oh, so the correct position is not initialized until we call eviction scan for the first time. Then the comment in Bucket.h is a bit confusing, as it implies we get a valid stream without the need to re-seek. So maybe clarify that in the comment?